programs/drift/src/instructions/user.rs

```diff
@@ -1362,6 +1372,137 @@ pub fn handle_place_and_make_perp_order<'c: 'info, 'info>(
+    taker_order_params_message.verify_all_same_market_indexes()?;
+    let market_index = taker_order_params_message.market_index;
+    let taker_order_params = taker_order_params_message.swift_order_params;
+
+    let clock = &Clock::get()?;
+    controller::repeg::update_amm(market_index, perp_market_map, oracle_map, state, clock)?;
+
+    // Use the first matching taker order param
+    let matching_taker_order_params_ref = taker_order_params.iter().find(|order_params| {
+        order_params.order_type != OrderType::TriggerLimit
+            && order_params.order_type != OrderType::TriggerMarket
+    });
+
+    if matching_taker_order_params_ref.is_none() {
+        msg!("No matching taker order params found");
+        // Place any matching orders and non-matching orders
+        return Ok(false);
+    }
+
+    let matching_taker_order_params = *matching_taker_order_params_ref.unwrap();
+
+    // Place taker order
+    let taker_order_sent = controller::orders::place_swift_perp_order(
+        state,
+        taker,
+        taker_key,
+        perp_market_map,
+        spot_market_map,
+        oracle_map,
+        clock,
+        matching_taker_order_params,
+        PlaceOrderOptions::default(),
+    )?;
+
+    if taker_order_sent {
+        let non_matching_orders = taker_order_params
+            .iter()
+            .filter(|order_params| {
+                order_params.order_type == OrderType::TriggerLimit
+                    || order_params.order_type == OrderType::TriggerMarket
+            })
+            .collect::<Vec<_>>();
+        for swift_order_params in non_matching_orders {
+            controller::orders::place_perp_order(
+                state,
+                taker,
+                taker_key,
+                perp_market_map,
+                spot_market_map,
+                oracle_map,
+                clock,
+                swift_order_params.to_order_params(),
+                PlaceOrderOptions::default(),
+            )?;
+        }
+    } else {
+        msg!(
+            "Orders not placed due to taker order id mismatch: taker account next order id {}, order params expected next order id {}",
+            taker.next_order_id,
+            matching_taker_order_params.get_expected_order_id()?
+        );
+    }
+
+    Ok(taker_order_sent)
+}
+
+pub fn handle_place_swift_taker_order<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, PlaceSwiftTakerOrder<'info>>,
+    taker_order_params_bytes: Vec<u8>,
+    sig: [u8; 64],
+) -> Result<()> {
+    let taker_order_params_message: SwiftOrderParamsMessage =
+        SwiftOrderParamsMessage::deserialize(&mut &taker_order_params_bytes[..]).unwrap();
+    let state = &ctx.accounts.state;
+
+    let AccountMaps {
+        perp_market_map,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        &mut ctx.remaining_accounts.iter().peekable(),
+        &get_writable_perp_market_set(taker_order_params_message.market_index),
```

Comment:
not urgent but we we probably want this to work with spot and perp (you can get market type from order params)

Reply:
also, for placing order, you shouldnt need to make perp market writable